### PR TITLE
Pootle config command

### PR DESCRIPTION
**Pull request by phlax**
Adds a `pootle config` command for getting/setting pootle_config vars

### CODE REVIEW

File: pootle/apps/pootle_app/management/commands/config.py [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+                else:
+                    model_query = dict(pk=kwargs["object"])
+                try:
+                    instance = ct_model.objects.get(**model_query)
+                except ct_model.DoesNotExist as e:
+                    raise CommandError(e)
+                except ct_model.MultipleObjectsReturned as e:
+                    raise CommandError(e)
+                except FieldError as e:
+                    raise CommandError(e)
```
Comment (unho) [inline pootle/apps/pootle_app/management/commands/config.py]: You are doing the same thing for different exceptions. Just wondering if it is better to combine them.

File: pootle/apps/pootle_app/management/commands/config.py [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+                return config.get(ct_model, instance=instance)
+            return config.get(ct_model)
+        return config.get()
+
+    def print_no_config(self, **kwargs):
+        self.stdout.write("No configuration found")
```
Comment (unho) [inline pootle/apps/pootle_app/management/commands/config.py]: Not localized string.

File: pootle/apps/pootle_app/management/commands/config.py (hunk 1 above)
Comment (phlax) [inline pootle/apps/pootle_app/management/commands/config.py]: yep

File: pootle/apps/pootle_app/management/commands/config.py [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+            dest="append",
+            metavar=("KEY", "VALUE"),
+            nargs=2,
+            help="Append config with this value")
+        parser.add_argument(
+            '-c', '--clear',
+            action='append',
+            dest='clear',
+            metavar="KEY",
+            help="Clear config with this key")
```
Comment (unho) [inline pootle/apps/pootle_app/management/commands/config.py]: The help messages are not localized.

File: tests/commands/config.py [hunk 4]
```diff
@@ -0,0 +1,785 @@
...
+    if not isinstance(value, (str, unicode)):
+        value_class = type(value).__name__
+        return (
+            "%s(%s)"
+            % (value_class,
+               json.dumps(value)))
+    return value
+
+
+def _test_config_get(out, key, model=None, instance=None, asjson=False):
```
Comment (unho) [inline tests/commands/config.py]: `asjson`?

Comment (phlax) [inline tests/commands/config.py]: not sure i follow. The command has 2 modes string mode and json mode, this lets the test flip between

File: tests/commands/config.py [hunk 5]
```diff
@@ -0,0 +1,785 @@
...
+    expected = ""
+    conf = config.get(model, instance=instance)
+    try:
+        expected = conf.get_config(key)
+        expected_class = type(expected).__name__
+        expected = json.dumps(expected)
+        if not asjson:
+            expected = "%s(%s)" % (expected_class, expected)
```
Comment (phlax) [inline tests/commands/config.py]: in string mode it wraps the var with the type, to disambiguate between a string that happens to be valid json, from some actual json

File: pootle/apps/pootle_app/management/commands/config.py [hunk 6]
```diff
@@ -0,0 +1,236 @@
...
+        parser.add_argument(
+            '-l', '--list',
+            action='append',
+            dest='list',
+            metavar="KEY",
+            help=_(
+                "List config with this key, "
+                "can be specified multiple times"))
+        parser.add_argument(
+            '-s', '--set',
```
Comment (unho) [inline pootle/apps/pootle_app/management/commands/config.py]: `--get` and `--set` look like mutually exclusive options, so perhaps it is better to put them in a [argparse mutually exclusive group](https://docs.python.org/2/library/argparse.html#mutual-exclusion).

Comment (phlax) [inline pootle/apps/pootle_app/management/commands/config.py]: thanks, thats the solution i was looking for!

Comment (unho) [inline pootle/apps/pootle_app/management/commands/config.py]: `--append` and `--clear` might belong to that group too.

Comment (phlax) [inline pootle/apps/pootle_app/management/commands/config.py]: thanks, thats the solution i was looking for!

File: pootle/apps/pootle_app/management/commands/config.py [hunk 7]
```diff
@@ -0,0 +1,236 @@
...
+                    raise CommandError(e)
+                return config.get(ct_model, instance=instance)
+            return config.get(ct_model)
+        return config.get()
+
+    def print_no_config(self, **kwargs):
+        self.stdout.write(_("No configuration found"))
+
+    def check_incompatible_flags(self, **kwargs):
+        mutually_incompatible = ["get", "set", "append", "list", "clear"]
```
Comment (unho) [inline pootle/apps/pootle_app/management/commands/config.py]: mmm, just saw this after commenting above about using argparse mutual exclusive group.

Comment (phlax) [inline pootle/apps/pootle_app/management/commands/config.py]: yep, i will remove and use argparse properly

**Comment by unho** [comment]
Other than the mutual exclusive options change, which I strongly prefer to be done before landing, this lgtm.

File: pootle/apps/pootle_app/management/commands/config.py [hunk 8]
```diff
@@ -0,0 +1,224 @@
...
+                        object_field)
+                else:
+                    pk = item.object_pk
+                item_name = "%s[%s]" % (ct, pk)
+            else:
+                item_name = ct
+        else:
+            item_name = "Pootle"
+        self.stdout.write(
+            "{: <25} {: <25} {: <30}".format(
```
Comment (ta2-1) [inline pootle/apps/pootle_app/management/commands/config.py]: @phlax can you explain me what's going on here?

Comment (phlax) [inline pootle/apps/pootle_app/management/commands/config.py]: column formatted output, the cols could probs do with being made wider, perhaps dynamically

Comment (phlax) [inline pootle/apps/pootle_app/management/commands/config.py]: @ta2-1 making this dynamic now...

**Comment by ta2-1** [comment]
lgtm

**Pull request merged**